src/core/selinux-access.c

```diff
@@ -226,6 +226,7 @@ int mac_selinux_generic_access_check(
 
                 if (getfilecon_raw(path, &fcon) < 0) {
                         r = -errno;
+                        enforce = security_getenforce() > 0;
```

Comment:
Why re-evaluate the enforcing state here?
It was retrieved in line 200.
There will always be scenarios, where we compute the enforcing state via `security_getenforce` and before using it, the actual state will be changed asynchronously and we would use the *"wrong"* state.
I think the time passed between this line and line 200 is negligible.

Reply:
An alternative is to drop line 200 and only compute `enforce` in the three error branches where needed